Makefile

```diff
@@ -92,19 +81,78 @@ lint:
 misspell:
 	@output=`misspell -error $(ALL_DOC)` && echo misspell finished successfully || (echo misspell errors:\\n$$output && exit 1)
 
+# --------------------------
+#  CI
+# --------------------------
+
+# Adds license headers to files that are missing it, quiet tests
+# so full output is visible at a glance.
+.PHONY: precommit
+precommit: addlicense lint misspell test_quiet
+
+# Checks for the presence of required license headers, runs verbose
+# tests for complete information in CI job.
+.PHONY: presubmit
+presubmit: checklicense lint misspell test
+
+# --------------------------
+#  Build and Test 
+# --------------------------
+
+GO_BUILD_OUT ?= ./bin/otelopscol
 .PHONY: build
 build:
-	go build -tags=$(GO_TAGS) -o ./bin/$(OTELCOL_BINARY) $(LD_FLAGS) ./cmd/otelopscol
+	go build -tags=$(GO_BUILD_TAGS) -o $(GO_BUILD_OUT) $(LD_FLAGS) ./cmd/otelopscol
+
+OTELCOL_BINARY = google-cloud-metrics-agent_$(GOOS)_$(GOARCH)$(EXTENSION)
+.PHONY: build_full_name
+build_full_name:
+	go build -tags=$(GO_BUILD_TAGS) -o ./bin/$(OTELCOL_BINARY) $(LD_FLAGS) ./cmd/otelopscol
```

Comment:
This is essentially the same command as in the `build` target, but with a different binary name. I wonder if it makes sense to factor out the common command, e.g.:
```
.PHONY: build-impl
build-impl:
ifndef BINARY_OUT
	$(error "BINARY_OUT is undefined")
endif
	go build -tags=$(GO_BUILD_TAGS) -o $(BINARY_OUT) $(LD_FLAGS) ./cmd/otelopscol

GO_BUILD_OUT ?= ./bin/otelopscol
.PHONY: build
build:
	$(MAKE) BINARY_OUT=$(GO_BUILD_OUT) build-impl

OTELCOL_BINARY = google-cloud-metrics-agent_$(GOOS)_$(GOARCH)$(EXTENSION)
.PHONY: build_full_name
build_full_name:
	$(MAKE) BINARY_OUT=./bin/$(OTELCOL_BINARY) build-impl
```

Reply:
Did a version of this in [a580ef7](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/157/commits/a580ef727c331ca6e3c78d47744f713d0396b907)